src/features/utils/instrument-base.js

```diff
@@ -85,19 +85,23 @@ export class InstrumentBase extends FeatureBase {
     if (isWorkerScope) importLater()
     else onWindowLoad(() => importLater(), true)
   }
-}
-/**
+
+  /**
  * Make a determination if an aggregate class should even be imported
  * @param {string} featureName
  * @param {SessionEntity} session
  * @returns
  */
-function shouldImportAgg (featureName, session) {
+  shouldImportAgg (featureName, session) {
   // if this isnt the FIRST load of a session AND
   // we are not actively recording SR... DO NOT run the aggregator
   // session replay samples can only be decided on the first load of a session
   // session replays can continue if in progress
-  if (featureName === FEATURE_NAMES.sessionReplay) return !!session?.isNew || !!session?.state.sessionReplayActive
-  // todo -- add case like above for session trace
-  return true
+    if (featureName === FEATURE_NAMES.sessionReplay) {
+      if (getConfigurationValue(this.agentIdentifier, 'session_trace.enabled') === false) return false
```

Comment:
I think this a bad idea. Trace depends on replay already; don't introduce a cyclic relationship. They're becoming too coupled.

Reply:
Session replay literally cannot run if session trace is not included.  This is not optional

Reply:
Consulted with larger group, this will be fine for LP.  May need to circle back for GA.  Added a dependency check in `featureDependencies.js` as suggested.